sklearn/tests/test_naive_bayes.py

```diff
@@ -94,6 +94,14 @@ def test_gnb_sample_weight():
     assert_array_almost_equal(clf_dupl.sigma_, clf_sw.sigma_)
 
 
+def test_gnb_class_prior():
+    """Test whether the class prior override is properly used. """
+    clf = GaussianNB(class_prior=np.array([0.3, 0.7])).fit(X, y)
+    assert_array_almost_equal(clf.predict_proba([[-0.1, -0.1]]),
+                              np.array([[0.825303662161683,
+                                         0.174696337838317]]), 8)
```

Comment:
I would intentionally make the P(y = 2) large and test predict as well, for ex

    gnb = GaussianNB(class_prior=[0.01, 0.99])
    gnb.fit(X, y)
    gnb.predict(([[-0.1, -0.1]])